enumerable.rb

```diff
@@ -0,0 +1,188 @@
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    to_a.my_each do |num|
+      bool = yield num
+      bool ? arr.push(num) : ''
+    end
+    arr
+  end
+
+  def my_all?(arg = false)
+    if block_given?
+      to_a.my_each { |num| return false unless yield num }
+    elsif arg && to_a.length.positive?
+      if arg.class.to_s == 'Regexp'
+        to_a.my_each { |obj| return false unless arg.match? obj.to_s }
+      elsif arg.class.to_s == 'Class'
+        to_a.my_each { |obj| return false unless obj.is_a? arg }
+      else
+        to_a.my_each { |obj| return false unless obj == arg }
+      end
+    elsif to_a.length.positive?
+      to_a.my_each { |obj| return false unless obj }
+    end
+    true
+  end
+
+  def my_any?(arg = false)
+    if block_given?
+      to_a.my_each { |num| return true if yield num }
+    elsif arg && to_a.length.positive?
+      if arg.class.to_s == 'Regexp'
+        to_a.my_each { |obj| return true if arg.match? obj.to_s }
+      elsif arg.class.to_s == 'Class'
+        to_a.my_each { |obj| return true if obj.is_a? arg }
+      else
+        to_a.my_each { |obj| return true if obj == arg }
+      end
+    elsif to_a.length.positive?
+      to_a.my_each { |obj| return true if obj }
+    end
+    false
+  end
+
+  def my_none?(arg = false)
+    if block_given?
+      to_a.my_each { |num| return false if yield num }
+    elsif arg && to_a.length.positive?
+      if arg.class.to_s == 'Regexp'
+        to_a.my_each { |obj| return false if arg.match? obj.to_s }
+      elsif arg.class.to_s == 'Class'
+        to_a.my_each { |obj| return false if obj.is_a? arg }
+      else
+        to_a.my_each { |obj| return false if obj == arg }
+      end
+    elsif to_a.length.positive?
+      to_a.my_each { |obj| return false if obj }
+    end
+    true
+  end
+
+  def my_count(arg = false)
+    if block_given?
+      return 'Error: arg and block not allowed at the same time' if arg
+
+      count = 0
+      to_a.my_each do |item|
+        count += 1 if yield item
+      end
+    else
+      return length unless arg
+
+      count = 0
+      to_a.my_each do |item|
+        count += 1 if item == arg
+      end
+      count
+    end
+    count
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    arr = []
+    to_a.my_each do |item|
+      arr.push(yield item)
+    end
+    arr
+  end
+
+  def my_inject(*args)
+    if args && !block_given?
+      if args.length == 2 && args[1].class.to_s == 'Symbol'
+        acc = args[0]
+        (0...to_a.length).my_each { |i| acc = acc.send(args[1].to_s, to_a[i]) }
+      elsif args[0].class.to_s == 'Symbol'
+        acc = to_a[0]
+        (1...to_a.length).my_each { |i| acc = acc.send(args[0].to_s, to_a[i]) }
+      else
+        acc = args
+        (0...to_a.length).my_each { |i| acc = yield(acc, to_a[i]) }
+      end
+    else
+      acc = to_a[0]
+      (1...to_a.length).my_each { |i| acc = yield(acc, to_a[i]) }
+    end
+    acc
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
This is what I get after testing that method:
![inject](https://user-images.githubusercontent.com/19773024/71774779-c7f37b00-2f85-11ea-896a-77691dc32730.png)

Reply:
How do I fix this? It returns the same thing as Ruby's inject

Reply:
@ClintonEnyinna 
You can do something like this;
`def my_inject(arg_1 = nil, arg_2 = nil, &block)
    array = to_a.dup
    (inject, sym, array) = get_block_and_symbol(arg_1, arg_2, array, block_given?)
    inject = if sym
               symbol(array, sym, inject)
             else
               block(array, inject, &block)
             end
    inject
  end`

Helper methods;
`def get_block_and_symbol(arg1, arg2, arr, block)
    arg1 = arr.shift if arg1.nil? && block
    return [arg1, nil, arr] if block
    return [arr.shift, arg1, arr] if arg2.nil?
    [arg1, arg2, arr]
  end`

`def symbol(array, sym, inject)
    array[0..-1].my_each { |i| inject = inject.send(sym, i) }
    inject
  end`

`def block(array, inject, &block)
    raise LocalJumpError, 'no block given' unless block
    array[0..-1].my_each { |i| inject = block.yield(inject, i) }
    inject
  end`

That should fix the problem.